From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrey Konovalov <andreyknvl@google.com>
Date: Mon, 15 Feb 2021 19:07:58 +0100
Subject: ANDROID: build_config: drop CONFIG_KASAN_PANIC_ON_WARN

CONFIG_KASAN_PANIC_ON_WARN was added in a custom patch for Pixel kernels,
which would make KASAN panic the kernel after the first report regardless
of whether the panic_on_warn parameter is set. (Coincidentally, that patch
also would break instrumentation mode selection for KASAN.)

As that patch was never applied to the common kernel,
CONFIG_KASAN_PANIC_ON_WARN doesn't exist here. This change drops the
non-existent CONFIG_KASAN_PANIC_ON_WARN from build.config.gki_kasan.

Signed-off-by: Andrey Konovalov <andreyknvl@google.com>
Change-Id: I9f42bb5f3515f18e2a5774241ea73a59d8883955
---
 build.config.gki_kasan | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.config.gki_kasan b/build.config.gki_kasan
--- a/build.config.gki_kasan
+++ b/build.config.gki_kasan
@@ -5,7 +5,6 @@ function update_kasan_config() {
     ${KERNEL_DIR}/scripts/config --file ${OUT_DIR}/.config \
          -e CONFIG_KASAN \
          -e CONFIG_KASAN_INLINE \
-         -e CONFIG_KASAN_PANIC_ON_WARN \
          -e CONFIG_KCOV \
          -e CONFIG_PANIC_ON_WARN_DEFAULT_ENABLE \
          -d CONFIG_RANDOMIZE_BASE \
